api/api_test.go

```diff
@@ -29,3 +37,94 @@ func testHTTPServer(
 
 	return config, ln
 }
+
+// nextPort is the next port to use for the API server.
+var nextPort int32 = 28200
+
+// restVaultServer runs an instance of the Vault server in development mode.
+// This requires that the vault binary is installed and in the $PATH.
+func testVaultServer(t *testing.T) (*Client, func()) {
```

Comment:
We followed a similar pattern in Consul - the main reason to use this style was to avoid a cyclic dependency. If TestCluster doesn't create one, then great! Maybe we can use that instead.

For actually running these tests, what we did with Consul was to just compile the "dev" target prior to running the "test" target in the Makefile. That way we could easily launch the binary from `$PWD/bin/vault` using the current HEAD.